console

```diff
@@ -1,31 +1,62 @@
 #!/bin/bash
 
+CALLING_DIR=$(pwd)
+
 function help {
 	echo "Remote Environment client."
 	echo
 	echo "Usage:"
 	echo "  "$0" command"
 	echo 
 	echo "Available commands:"
-	echo "   setup             Setup the remote environment client"
+	echo "   setup             Setup the remote environment client and settings"
+	echo "   setup:client      Setup the remote environment client"
+	echo "   setup:remote      Set the remote settings"
 	echo "   create            Create the remote environment"
 	echo "   watch             Watch local changes and synchronize with the remote environment"
 	echo "   ssh               SSH into the remote environment container"
-	echo "   build             Update the remove environment with local changes"
+	echo "   build             Update the remote environment with local changes"
 	echo "   destroy           Destroy the remote environment"
 }
 
 command_exists () {
     type "$1" &> /dev/null ;
 }
 
-function get_current_namespace()
-{
-    local  myresult='some value'
-    echo "$myresult"
+function dir {
+    echo $CALLING_DIR
 }
 
-function setup {
+function remote_name {
+    echo $(read_setting 3)
+}
+
+function remote_branch {
+    echo $(read_setting 2)
+}
+
+function namespace {
+    echo $(read_setting 1)
+}
+
+function pod {
+    echo `kubectl --namespace=$(namespace) get pods | grep $1 | awk '{print $1}'`
```

Comment:
That method is doing an API call to the cluster's API every time. The name of the pod changes only every deployment, so I'm sure we could "cache" it until there is an exception somewhere. That would improve A LOT the synchronization speed.